### PR TITLE
Batch subscribe

### LINKED ISSUE #708: Add batch subscribe relay method

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
- Add batch subscribe relay method
- Integrate batch subscribe into resubscription services
- change socket connection publisher subject to current value subject; in sample wallet sessionEngine was usually initialised after socket was already connected and Sign SDK's resubscribtion was never triggered.

Resolves #708 

## How Has This Been Tested?
- manually

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

**Review by alexander-lsvk — APPROVED** [review]

File: Sources/Chat/ProtocolServices/Common/ResubscriptionService.swift [hunk 1]
```diff
@@ -32,6 +32,7 @@ class ResubscriptionService {
 
     func resubscribe(account: Account) async throws {
         let topics = chatStorage.getThreads(account: account).map { $0.topic }
+        guard !topics.isEmpty else { return }
```
Comment (flypaper0) [inline Sources/Chat/ProtocolServices/Common/ResubscriptionService.swift:35]: may be do this check inside butchSubscribe?

File: Sources/WalletConnectRelay/RelayClient.swift [hunk 2]
```diff
@@ -189,6 +193,38 @@ public final class RelayClient {
...
 
+    public func batchSubscribe(topics: [String]) async throws {
+        guard !topics.isEmpty else { throw Errors.batchContainsNoItems }
+        logger.debug("Relay: Subscribing to topics: \(topics)")
+        let rpc = BatchSubscribe(params: .init(topics: topics))
+        let request = rpc
+            .asRPCRequest()
+        let message = try! request.asJSONEncodedString()
+        var cancellable: AnyCancellable?
+        cancellable = batchSubscriptionResponsePublisher
```
Comment (flypaper0) [inline Sources/WalletConnectRelay/RelayClient.swift:204]: we are doing the same logic in different places. Could we refactor it somehow?

File: Sources/WalletConnectRelay/RelayClient.swift [hunk 3]
```diff
@@ -292,6 +317,8 @@ public final class RelayClient {
                     requestAcknowledgePublisherSubject.send(response.id)
                 } else if let subscriptionId = try? anyCodable.get(String.self) {
                     subscriptionResponsePublisherSubject.send((response.id, subscriptionId))
+                } else if let subscriptionIds = try? anyCodable.get([String].self) {
+                    batchSubscriptionResponsePublisherSubject.send((response.id, subscriptionIds))
```
Comment (flypaper0) [inline Sources/WalletConnectRelay/RelayClient.swift:321]: do we need to use different ack publishers? Can we use one?

Comment (llbartekll) [reply Sources/WalletConnectRelay/RelayClient.swift:321]: 👍 we can

File: Sources/Chat/ProtocolServices/Common/ResubscriptionService.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/Chat/ProtocolServices/Common/ResubscriptionService.swift:35]: we do but it throws an error batchContainsNoItems

**Review by flypaper0 — APPROVED** [review]

**Pull request merged**